mozci/push.py

```diff
@@ -1132,27 +1131,46 @@ def __make_group_summary_objects(pushes):
     """
     # Obtain all the task.id values contained in the pushes. It will be used in
     # the `where` query against ActiveData.
-    task_ids = [task.id for push in pushes for task in push.tasks]
-
-    # Extract the results of the above query.
-    results = run_query("group_durations", Namespace(task_id=task_ids))["data"]
+    results = []
+    revs = [p.rev for p in pushes]
+    start = 0
+    revcount = 0
+    try:
+        # if we have too many revsisions, activedata returns an error
+        while revcount < len(revs):
+            revcount += 30
+            if revcount > len(revs):
+                revcount = len(revs)
+            results += run_query("group_durations", Namespace(push_ids=revs[start:revcount], branch=branch))["data"]
+            start = revcount
+    except MissingDataError:
+        pass
 
     # Sort by the result.group attribute.
     results = sorted(results, key=lambda x: x[1])
 
-    # Dictionary to hold the mapping keyed by result.group mapped to list of
-    # task.id and list of and result.duration.
-    mapping = defaultdict(lambda: defaultdict(list))
+    tasks_by_config = {}
+    task_id_to_task = {}
+    for push in pushes:
+        for t in push.tasks:
+            task_id_to_task[t.id] = t
```

Comment:
Nit: you can use a dict comprehension: `task_id_to_task = {t.id: t for push in pushes for t in push.tasks}`